lightning/src/blinded_path/message.rs

```diff
@@ -241,7 +241,10 @@ pub(crate) struct ReceiveTlvs {
 	/// If `context` is `Some`, it is used to identify the blinded path that this onion message is
 	/// sending to. This is useful for receivers to check that said blinded path is being used in
 	/// the right context.
-	pub context: Option<MessageContext>
+	pub context: Option<MessageContext>,
+
+	/// Custom Tlvs. A user can use this to send custom tlvs information back to themself.
+	pub custom_tlvs: Option<Vec<u8>>,
```

Comment:
Seems like this is just custom data rather than custom TLVs. Nothing is enforcing that this data is encoded as a TLV stream.

@TheBlueMatt Not sure if you have any thoughts on how if we should make this a `Vec<(u64, Vec<u8>)>` instead? And, if so, if we should encode it inside type `65539` or place them directly in `ReceiveTlvs`, restricting TLVs >= 2^16 and all the checks that that entails.

Reply:
For now, I've updated the documentation to include a clear note on Forward Compatibility.
Let me know if we prefer moving to the `Vec<(u64, Vec<u8>)>` approach—I’d would love to integrate that!